wagtail_live/models.py

```diff
@@ -59,6 +58,168 @@ class LivePageMixin(models.Model):
+        """
+
+        try:
+            live_post = self.live_posts[live_post_index]
+        except KeyError:
+            raise 
+        return live_post
+
+    def get_live_post_by_id(self, live_post_id):
+        """Retrieve a live post by its ID.
+        
+        Args: 
+            live_post_id (str): ID of the live post to look for.
+
+        Returns:
+            (LivePostBlock) The live post instance
+
+        Raises:
+            (someException) if a live post with the given ID doesn't exist.
+        """
+
+        live_post_index = self.get_live_post_index(live_post_id=live_post_id)
+        if live_post_index == -1:
+            raise
+        return self.get_live_post_by_index(live_post_index)
+
+    def add_block_to_live_post(self, block_type, block, live_block):
+        """ Add a new content block to a live post.
+        
+        Args:
+            block_type (str):
+                Type of the block to add
+            block (Block):
+                Block to add to the live post.
+            live_block (LivePostBlock):
+                Live post in which the new block will be added.
+        """
+
+        live_block["content"].append((block_type, block))
+
+    def add_live_post(self, live_post, live_post_id):
+        """ Add a new live post to live page.
+
+        Live posts are added with a custom ID to facilitate
+        keeping their track. 
+        By default, the value of a live post's ID is
+        the timestamp of the corresponding message creation. 
+        A check is also done to ensure that live posts are always
+        sorted by time.
+
+        Args:
+            live_post (LivePostBlock):
+                live post to add
+            live_post_id (str):
+                Id of the new live_post
+        """
+
+        lp_index = len(self.live_posts) if self.live_posts else 0
+        post_id = float(live_post_id)
+        while self.live_posts and float(self.live_posts[lp_index - 1].id) > post_id:
+            lp_index -= 1
+
+        # Insert to keep posts sorted by time
+        self.live_posts.insert(lp_index, ("live_post", live_post, live_post_id))
+
+        self.last_update_at = now()
+        self.save()
+
+    def delete_live_post(self, live_post_id):
+        """Delete a live post by its ID.
+        
+        Args:
+            live_post_id (str): Id of the live post to delete.
+
+        Raises:
+            (someException) if live post with the given ID doesn't exist.
+        """
+
+        live_post_index = self.get_live_post_index(live_post_id=live_post_id)
+        if live_post_index == -1:
+            raise
+        del self.live_posts[live_post_index]
+        self.last_update_at = now()
+        self.save()
+
+    def clear_live_post_content(self, live_post):
+        """Clears the content of a live post.
+
+        Args:
+            live_post (livePostBlock): Live post which content will be cleared.
+        """
+    
+        live_post.value.get("content").clear()
```

Comment:
If you assume you will get a valid value (by immediately calling `clear` afterwards), I'd use `live_post.value['content']` over `live_post.get('content')`.

Reply:
Ok, I will do that.

Reply:
And why would you do that?

Reply:
Because if 'content' is not present, with `value['content']` a `KeyError` would be raised saying `'content'` doesn't exist, whereas when using `value.get('content').clear()` an `AttributeError` would be raised saying None type has no attribute `clear`. Well it doesn't matter that much, but the first one seems more clear to me.

Reply:
Oh ok. Yes, the first one is clearer yes.